Specs/Scene/ImplicitTileCoordinatesSpec.js

```diff
@@ -39,24 +61,347 @@ describe("Scene/ImplicitTileCoordinates", function () {
+        .getAncestorCoordinates(0)
+        .isEqual(internalQuadtree(0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalQuadtree(1, 0, 0)
+        .getAncestorCoordinates(1)
+        .isEqual(internalQuadtree(0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalQuadtree(1, 1, 1)
+        .getAncestorCoordinates(1)
+        .isEqual(internalQuadtree(0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalQuadtree(2, 3, 3)
+        .getAncestorCoordinates(1)
+        .isEqual(internalQuadtree(1, 1, 1))
+    ).toEqual(true);
+
+    expect(
+      internalQuadtree(2, 3, 3)
+        .getAncestorCoordinates(2)
+        .isEqual(internalQuadtree(0, 0, 0))
+    ).toEqual(true);
+  });
+
+  it("getAncestorCoordinates works as expected for octree", function () {
+    expect(
+      internalOctree(0, 0, 0, 0)
+        .getAncestorCoordinates(0)
+        .isEqual(internalOctree(0, 0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalOctree(1, 0, 0, 0)
+        .getAncestorCoordinates(1)
+        .isEqual(internalOctree(0, 0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalOctree(1, 1, 1, 1)
+        .getAncestorCoordinates(1)
+        .isEqual(internalOctree(0, 0, 0, 0))
+    ).toEqual(true);
+
+    expect(
+      internalOctree(2, 3, 3, 3)
+        .getAncestorCoordinates(1)
+        .isEqual(internalOctree(1, 1, 1, 1))
+    ).toEqual(true);
+
+    expect(
+      internalOctree(2, 3, 3, 3)
+        .getAncestorCoordinates(2)
+        .isEqual(internalOctree(0, 0, 0, 0))
+    ).toEqual(true);
+  });
+
+  it("getOffsetCoordinates throws with invalid inputs", function () {
+    // undefined input
+    expect(function () {
+      return internalQuadtree(0, 0, 0).getOffsetCoordinates(undefined);
+    }).toThrowDeveloperError();
+
+    // descendant is above ancestor
+    expect(function () {
+      return internalQuadtree(1, 0, 0).getOffsetCoordinates(
+        internalQuadtree(0, 0, 0)
+      );
+    }).toThrowDeveloperError();
+
+    // descendant is not actually a descendant
+    expect(function () {
+      return internalQuadtree(1, 0, 0).getOffsetCoordinates(
+        internalQuadtree(2, 3, 3)
+      );
+    }).toThrowDeveloperError();
+
+    // mismatched subdivisionScheme
+    expect(function () {
+      return new ImplicitTileCoordinates({
+        subdivisionScheme: ImplicitSubdivisionScheme.QUADTREE,
+        subtreeLevels: 2,
+        level: 0,
+        x: 0,
+        y: 0,
+      }).getOffsetCoordinates(
+        new ImplicitTileCoordinates({
+          subdivisionScheme: ImplicitSubdivisionScheme.OCTREE,
+          subtreeLevels: 2,
+          level: 0,
+          x: 0,
+          y: 0,
+          z: 0,
+        })
+      );
+    }).toThrowDeveloperError();
+
+    // mismatched subtreeLevels
+    expect(function () {
+      return new ImplicitTileCoordinates({
+        subdivisionScheme: ImplicitSubdivisionScheme.QUADTREE,
+        subtreeLevels: 2,
+        level: 0,
+        x: 0,
+        y: 0,
```

Comment:
Is there a reason why `internalQuadtree()` isn't used here like in many of the tests above? I do like the concise helper function